### PR TITLE
Migrate from NAN to N-API

### PR DESCRIPTION
**Pull request by melikhov-dev**

### CODE REVIEW

File: src/time.cc [hunk 1]
```diff
@@ -2,109 +2,117 @@
 #include <stdint.h>
 #include <time.h>
 
-#include "nan.h"
+#include "napi.h"
+#include "uv.h"
```
Comment (tchetwin) [inline src/time.cc]: Is this used?

File: src/time.cc [hunk 2]
```diff
@@ -2,109 +2,117 @@
...
+#include "math.h"
 
-using namespace node;
-using namespace v8;
+using namespace Napi;
```
Comment (tchetwin) [inline src/time.cc]: Is this needed? I think everything is using the namespace explicitly.

File: src/time.cc [hunk 3]
```diff
@@ -118,43 +126,45 @@ class Time {
...
 
 extern "C" {
-  static void init (Handle<Object> target) {
-    Time::Init(target);
+  static Napi::Object init (Napi::Env env, Napi::Object exports) {
+    return Time::Init(env, exports);
   }
-  NODE_MODULE(time, init)
+  NODE_API_MODULE(time, init)
 }
```
Comment (tchetwin) [inline src/time.cc:167]: I'm not sure that we need this `extern` block any more, or the wrapper around `Time::Init`

```cpp
NODE_API_MODULE(time, Time::Init);
```

File: src/time.cc [hunk 4]
```diff
@@ -2,109 +2,117 @@
 #include <stdint.h>
 #include <time.h>
 
-#include "nan.h"
+#include "napi.h"
+#include "uv.h"
+#include "math.h"
```
Comment (tchetwin) [inline src/time.cc]: [cmath](https://en.cppreference.com/w/cpp/header/cmath) header?

File: src/time.cc (hunk 3 above)
Comment (melikhov-dev) [reply src/time.cc:167]: But macro NODE_API_MODULE doesn't work with namespace in regfunc  https://github.com/nodejs/node-addon-api/issues/249

File: src/time.cc (hunk 1 above)
Comment (melikhov-dev) [reply src/time.cc]: nope 9b6f5f21fccd6b4a432f9fb0ba4b6750436dc49f

File: src/time.cc (hunk 2 above)
Comment (melikhov-dev) [reply src/time.cc]: I fixied it 9b6f5f21fccd6b4a432f9fb0ba4b6750436dc49f

File: src/time.cc (hunk 4 above)
Comment (melikhov-dev) [reply src/time.cc]: Thanks! 9b6f5f21fccd6b4a432f9fb0ba4b6750436dc49f

File: package-lock.json [hunk 5]
```diff
@@ -0,0 +1,215 @@
+{
```
Comment (gabrielschulhof) [inline package-lock.json]: Adding this file to the repo is unrelated to a port to N-API. If adding a `package-lock.json` is desired, it should be done in a separate PR.

Comment (gabrielschulhof) [reply package-lock.json]: Thus, IMO this file should be removed from this PR.

Comment (melikhov-dev) [reply package-lock.json]: Done 95fea6ad9493fed11dd72c709d73273c07d0fd29
